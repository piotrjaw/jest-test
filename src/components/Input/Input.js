import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Input extends PureComponent {
  static propTypes = {
    onChange: PropTypes.func.isRequired
  };

  state = { value: '' };

  clear = () => this.setState(
    () => ({ value: '' }),
    () => this.props.onChange({ target: { value: '' } })
  );

  handleChange = (e) => {
    e.persist();
    this.setState(
      () => ({ value: e.target.value }),
      () => this.props.onChange(e)
    );
  }

  render() {
    return (
      <input
        type="text"
        value={ this.state.value }
        onChange={ this.handleChange }
      />
    );
  }
}
