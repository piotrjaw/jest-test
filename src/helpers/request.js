import axios from 'axios';

const contructUrl = (url) => `http://localhost:5000${url}`;

export const del = (url) => axios.delete(contructUrl(url));

export const get = (url) => axios.get(contructUrl(url));

export const post = (url, data) => axios.post(contructUrl(url), data);

export const put = (url, data) => axios.put(contructUrl(url), data);
