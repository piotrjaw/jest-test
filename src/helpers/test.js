import React from 'react';

export const getWrapper = ({ Component, method, props }) => {
  return method(
    <Component
      { ...props }
    >
      { props.children }
    </Component>
  );
};

export const getMockStore = (mockFn, state = {}) => ({
  subscribe: mockFn(),
  dispatch: mockFn(),
  getState: mockFn(() => ({
    ...state,
  })),
});
