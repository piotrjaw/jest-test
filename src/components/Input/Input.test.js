import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getWrapper } from '../../helpers/test';
import Input from './Input';

const defaultProps = {
  onChange: jest.fn()
};

const setupWrapper = (config = {}) => {
  const { method = shallow, props = {} } = config;
  return getWrapper({
    Component: Input,
    method,
    props: {
      ...defaultProps,
      ...props
    }
  });
};

let wrapper;

beforeEach(() => {
  wrapper = setupWrapper();
});

afterEach(() => {
  expect(toJSON(wrapper.render())).toMatchSnapshot();
  wrapper = null;
  jest.resetAllMocks();
});

describe('Input component', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should handle onChange event', () => {
    const event = {
      target: { value: 'test' },
      persist: jest.fn()
    };
    wrapper.find('input').simulate('change', event);
    expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
    expect(wrapper.state()).toEqual({ value: 'test' });
    expect(event.persist).toHaveBeenCalledTimes(1);
  });

  it('should handle clear call', () => {
    const event = {
      target: { value: 'test' },
      persist: jest.fn()
    };
    wrapper.find('input').simulate('change', event);
    expect(wrapper.state()).toEqual({ value: 'test' });
    expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
    wrapper.instance().clear();
    expect(wrapper.state()).toEqual({ value: '' });
    expect(defaultProps.onChange).toHaveBeenCalledTimes(2);
  });
});
