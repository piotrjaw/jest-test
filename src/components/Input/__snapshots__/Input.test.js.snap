// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should handle clear call 1`] = `
<input
  type="text"
  value=""
/>
`;

exports[`Input component should handle onChange event 1`] = `
<input
  type="text"
  value="test"
/>
`;

exports[`Input component should render 1`] = `
<input
  type="text"
  value=""
/>
`;
